frame/example-offchain-worker/src/lib.rs

```diff
@@ -331,35 +382,28 @@ impl<T: Trait> Module<T> {
 
 	/// A helper function to fetch the price and send signed transaction.
 	fn fetch_price_and_send_signed() -> Result<(), &'static str> {
-		use system::offchain::SubmitSignedTransaction;
-		// Firstly we check if there are any accounts in the local keystore that are capable of
-		// signing the transaction.
-		// If not it doesn't even make sense to make external HTTP requests, since we won't be able
-		// to put the results back on-chain.
-		if !T::SubmitSignedTransaction::can_sign() {
```

Comment:
This check is missing, we should still show case how to avoid making unnecessary logic (fetching price) when there are no local accounts that can be used for signing.

Reply:
I introduced `can_sign` to `Signer` and re-added this check.